v0/destinations/delighted/transform.js

```diff
@@ -0,0 +1,238 @@
+const { EventType } = require("../../../constants");
+const {
+  getFieldValueFromMessage,
+  CustomError,
+  defaultRequestConfig,
+  extractCustomFields,
+  removeUndefinedAndNullValues,
+  getErrorRespEvents,
+  constructPayload,
+  getSuccessRespEvents,
+  getDestinationExternalID,
+  isEmptyObject,
+  defaultPostRequestConfig,
+  getValueFromMessage
+} = require("../../util");
+const logger = require("../../../logger");
+const {
+  isValidUserIdOrError,
+  eventValidity,
+  isValidEmail,
+  isValidPhone,
+  userValidity
+} = require("./util");
+const {
+  ENDPOINT,
+  DELIGHTED_EXCLUSION_FIELDS,
+  identifyMapping
+} = require("./config");
+
+const identifyResponseBuilder = (message, { Config }) => {
+  const userId = getFieldValueFromMessage(message, "userIdOnly");
+  if (!userId) {
+    throw new CustomError("userId is required for identify", 400);
+  }
+  let channel =
+    getDestinationExternalID(message, "delightedChannelType") || Config.channel;
+  channel = channel.toLowerCase();
+
+  isValidUserIdOrError(channel, userId);
+  let payload = constructPayload(message, identifyMapping);
+  
+  if(channel === "email"){
+    payload.email = userId;
+    if(payload.phone_number && !isValidPhone(payload.phone_number)){
+      payload.phone_number= null;
+      logger.error("Phone number format must be E.164.");
+  }
+  }else if(channel == "phone"){
+    payload.phone_number = userId;
+    if(payload.email && !isValidEmail(payload.email)){
+      payload.email = null;
+      logger.error("Email format is not correct.");
+    }
+  }
+  
+  payload.send = "false";
+  payload.channel = channel;
+  payload.delay = Config.delay || 0;
+  if (!payload.name) {
+    const fName = getFieldValueFromMessage(message, "firstName").trim();
+    const lName = getFieldValueFromMessage(message, "lastName").trim();
+    const name = fName.concat(" ").concat(lName);
+    if (name) {
+      payload.name = name;
+    }
+  }
+  let properties = {};
+  properties = extractCustomFields(
+    message,
+    properties,
+    ["traits", "context.traits"],
+    DELIGHTED_EXCLUSION_FIELDS
+  );
+  if(!isEmptyObject(properties)){
+    payload = {
+      ...payload,
+      properties
+    };
+  }
+  let lastSentAt= getValueFromMessage(message,"lastSentAt");
+  if(lastSentAt !== undefined && lastSentAt !== null)
```

Comment:
we can use `if (!lastSentAt)`
also lets try to maintain the formatting like for instace `if() {}`